src/stores/search.ts

```diff
@@ -0,0 +1,341 @@
+    ][]
+    return filterEntries.some(
+      ([filterKey, filterItems]) =>
+        filterKey !== 'mature' && filterItems.some((filter) => filter.checked)
+    )
+  })
+
+  // Actions
+  /**
+   * After a search type is changed, unchecks all the filters that are not
+   * applicable for this Media type.
+   */
+  function clearOtherMediaTypeFilters(searchType: SupportedSearchType) {
+    const mediaTypesToClear = supportedSearchTypes.filter(
+      (type) => type !== searchType
+    )
+    const filterKeysToClear = mediaTypesToClear.reduce((acc, type) => {
+      return [...acc, ...mediaUniqueFilterKeys[type]]
+    }, [] as FilterCategory[])
+
+    allFilterCategories.forEach((filterCategory) => {
+      if (filterKeysToClear.includes(filterCategory)) {
+        state.filters[filterCategory] = state.filters[filterCategory].map(
+          (f) => ({
+            ...f,
+            checked: false,
+          })
+        )
+      }
+    })
+  }
+
+  /**
+   * Replaces filters with the newFilterData parameter, making sure that
+   * audio/image provider filters are handled correctly.
+   */
+  function replaceFilters(newFilterData: Filters) {
+    const providerFilters: FilterCategory[] = [
+      'audioProviders',
+      'imageProviders',
+    ]
+
+    allFilterCategories.forEach((filterCategory) => {
+      if (providerFilters.includes(filterCategory)) {
+        newFilterData[filterCategory].forEach((provider) => {
+          const idx = state.filters[filterCategory].findIndex(
+            (p) => p.code === provider.code
+          )
+          if (idx > -1) {
+            state.filters[filterCategory][idx].checked = provider.checked
+          }
+        })
+      } else {
+        state.filters[filterCategory] = newFilterData[filterCategory]
+      }
+    })
+  }
+
+  /**
+   * Merge providers from API response with the filters that came from the browse URL search query string
+   * and match the checked properties in the store.
+   */
+  function initProviderFilters({
+    mediaType,
+    providers,
+  }: {
+    mediaType: SupportedMediaType
+    providers: { source_name: string; display_name: string }[]
+  }) {
+    const providersKey: FilterCategory = `${mediaType}Providers`
+    const currentProviders = state.filters[providersKey]
+      ? [...state.filters[providersKey]]
+      : []
+    state.filters[providersKey] = providers.map((provider) => {
+      const existingProviderFilterIdx = currentProviders.findIndex(
+        (p) => p.code === provider.source_name
+      )
+      const checked =
+        existingProviderFilterIdx >= 0
+          ? currentProviders[existingProviderFilterIdx].checked
+          : false
+
+      return {
+        code: provider.source_name,
+        name: provider.display_name,
+        checked,
+      }
+    })
+  }
+  /**
+   * Toggles a filter's checked parameter. Requires either codeIdx or code.
+   */
+  function toggleFilter(params: {
+    filterType: import('../store/types').FilterCategory
```

Comment:
This is already imported anyway.

```suggestion
    filterType: FilterCategory
```